dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/ShowIdentifiersStep.java

```diff
@@ -0,0 +1,170 @@
+        // for the new item, and the DOI consumer (if also configured) will mint or delete DOI entries as appropriate
+        // while the item is saved in the submission / workflow process
+
+        // This step simply looks for existing identifier data and returns it in section data for rendering
+        return getIdentifierData(obj);
+    }
+
+    /**
+     * Get data about existing identifiers for this in-progress submission item - this method doesn't require
+     * submissionService or step config, so can be more easily called from doPatchProcessing as well
+     *
+     * @param obj   The workspace or workflow item
+     * @return      A simple DataIdentifiers bean containing doi, handle and list of other identifiers
+     */
+    private DataIdentifiers getIdentifierData(InProgressSubmission obj) {
+        Context context = getContext();
+        DataIdentifiers result = new DataIdentifiers();
+        // Load identifier service
+        IdentifierService identifierService =
+                IdentifierServiceFactory.getInstance().getIdentifierService();
+        // Attempt to look up handle and DOI identifiers for this item
+        String[] defaultTypes = {"handle", "doi"};
+        List<String> displayTypes = Arrays.asList(configurationService.getArrayProperty(
+                "identifiers.submission.display",
+                defaultTypes));
+        result.setDisplayTypes(displayTypes);
+        String handle = identifierService.lookup(context, obj.getItem(), Handle.class);
+        DOI doi = null;
+        String doiString = null;
+        try {
+            doi = IdentifierServiceFactory.getInstance().getDOIService().findDOIByDSpaceObject(context, obj.getItem());
+            if (doi != null && !DOIIdentifierProvider.MINTED.equals(doi.getStatus())
+                    && !DOIIdentifierProvider.DELETED.equals(doi.getStatus())) {
+                doiString = doi.getDoi();
+            }
+        } catch (SQLException e) {
+            log.error(e.getMessage());
+        }
+
+        // Other identifiers can be looked up / resolved through identifier service or
+        // its own specific service here
+
+        // If we got a DOI, format it to its external form
+        if (StringUtils.isNotEmpty(doiString)) {
+            try {
+                doiString = IdentifierServiceFactory.getInstance().getDOIService().DOIToExternalForm(doiString);
+            } catch (IdentifierException e) {
+                log.error("Error formatting DOI: " + doi);
+            }
+        }
+        // If we got a handle, format it to its canonical form
+        if (StringUtils.isNotEmpty(handle)) {
+            handle = HandleServiceFactory.getInstance().getHandleService().getCanonicalForm(handle);
+        }
+
+        // Populate bean with data and return, if the identifier type is configured for exposure
+        result.addIdentifier("doi", doiString,
```

Comment:
Is the intention here to always have a "doi" shown in this step *even if* the DOI is empty?  I noticed that this results in the JSON always including a DOI section even if it's empty.  For instance, if DOIs are disabled but I'm pre-registering Handles, I'm seeing an identifier section like this:
```
    "identifiers" : [ {
        "id" : null,
        "value" : null,
        "identifierType" : "doi",
        "identifierStatus" : null,
        "type" : "identifier"
      }, {
        "id" : "http://localhost:4000/handle/123456789/7183",
        "value" : "http://localhost:4000/handle/123456789/7183",
        "identifierType" : "handle",
        "identifierStatus" : null,
        "type" : "identifier"
      } ],
      "displayTypes" : [ "handle" ]
```

It just seems odd to me that we'd include a "doi" or a "handle" when they are *null*.  I'd expect us to have a `isNotEmpty()" check around both of these `.addIdentifier()` calls.

That said, this is really minor... if it's needed, we can leave it.  But, it just looks odd to me that we have null DOIs or Handles returned in this step.

Reply:
@tdonohue in this case, my intention was to establish a difference between "not getting a DOI" (null) and non-existant (not configured / no DOIs used)- it is so the section can explicitly say "i'm configured to display DOIs, but one will *not* be minted for this item as things currently stand".

Reply:
@kshepherd : I guess I was just surprised that *with DOIs disabled* (as is in my setup), I still see a `null` DOI listed.  I can understand though how this would be useful when DOIs are *enabled*.

That said, as I mentioned, this is super minor.  If it's better to leave this as-is, it's fine with me.  It just struck me as "odd" since I don't even have DOIs enabled

Reply:
yes, fair enough. i think there are several areas which can be improved in this already-large feature, i'll expand on that a bit in my final comment after pushing changes, but i hope that these "areas for improvement" won't be blockers -- we'll see!

Reply:
Nope, not a blocker. As I said, it's "odd", but it'd only be a blocker if it was *assigning* a DOI :)  It's just a random null DOI, which we could clean up later if we need to.  If there's an easy fix though, feel free.